test/wolt_modal_sheet_test.dart

```diff
@@ -318,4 +320,27 @@ void main() {
       expect(modal.isAtLastPage, isTrue);
     });
   });
+
+  testWidgets('WoltModalSheet.modalBarrierColor defaults', (tester) async {
+    Color barrierColor = Colors.black54;
```

Comment:
How about adding a static const field inside `WoltModalSheetThemeData` and using it here to assert? Currentlt we are hardcoding the value here but this test will fail when the default value in the implementation changes

Reply:
>  Currentlt we are hardcoding the value here but this test will fail when the default value in the implementation changes

That's the idea behind adding tests. The test should fail if you modify expected source code. In such case, it will be a breaking change and tests will be updated too.

Reply:
Adding static to the actual source code and then using the value directly in the test will defeat the whole purpose of the test. The testing is a safety net to make sure the source is independently verified and when the source code is modified the this safety will fail. 1. It prevent the source code from being modified. 2. When making disruptive changes the test is also updated changing the expect value.

Reply:
> Currentlt we are hardcoding the value here but this test will fail when the default value in the implementation changes

If the goal is to change implementation without failing any tests then we won't need tests in the first place.

Reply:
There is a chance i maybe misunderstood your comment.

Reply:
The tests should fail on making disruptive changes in the source code. Using a static const value directly in the tests will defeat the test purpose.

Reply:
It depends on what you are testing. If you are testing the default value itself yes, using static const in the source code and tests is bad. If you are testing that the default value should be taken whatever the value is, then I think tests should not fail.